app/Actions/Import/Exec.php

```diff
@@ -165,128 +165,134 @@ private function memWarningCheck()
 	 * @throws ImagickException
 	 */
 	public function do(
-		string $path,
+		string $path_list,
 		$albumID,
 		$ignore_list = null
 	) {
-		// Parse path
-		$origPath = $path;
-
-		if (!$this->parsePath($path, $origPath)) {
-			return;
-		}
-
-		// We process breadth-first: first all the files in a directory,
-		// then the subdirectories.  This way, if the process fails along the
-		// way, it's much easier for the user to figure out what was imported
-		// and what was not.
-
-		// Update ignore list
-		$ignore_list = $this->setUpIgnoreList($path, $ignore_list);
-
-		$files = glob($path . '/*');
-		$filesTotal = count($files);
-		$filesCount = 0;
-		$dirs = [];
-		$lastStatus = microtime(true);
-
-		// Add '%' at end for CLI output
-		$percent_symbol = ($this->statusCLIFormatting) ? '%' : '';
-
-		$this->status_progress($origPath, '0' . $percent_symbol);
-		foreach ($files as $file) {
-			// re-read session in case cancelling import was requested
-			session()->start();
-			if (Session::has('cancel')) {
-				Session::forget('cancel');
-				$this->status_error($origPath, 'Import cancelled');
-				Logs::warning(__METHOD__, __LINE__, 'Import cancelled');
-
+		
+		$paths = explode(',', $path_list);
```

Comment:
```suggestion
		$paths = explode(';', $path_list);
```

As @ildyria said, `;` would be a much more common choice.

Reply:
Hi @nagmat84, @ildyria originally suggested a comma: "Comma separated is usually a more natural option."

Reply:
OK, maybe, then I got that wrong. Anyway, I would expect either an array as input parameter or even call the method iteratively with a single path. See my  comment below wrt. to line 168.

Reply:
As per [this](https://superuser.com/a/748264), I would say screw you if you use comma or semi-colon in your path. :smile:

Reply:
Thanks for your input folks, I will get onto this but may be a while, will chip away at weekends, got hands full with a newborn at moment - Lychee is proving great for sharing the photos with the family.

Reply:
@ildyria Awful source aside, well designed tools allow escaping.